### PR TITLE
[Editor]: Rework action-menu

### PR DESCRIPTION
**Pull request by cmoinier**
### Description

This PR reworks the UI of the action-menu in the dashboard.
It now uses the Angular CDK Overlay like in the [import button](https://github.com/geonetwork/geonetwork-ui/pull/985).

### Screenshots

Screenshot from 2024-10-24 15-17-47

image

### Quality Assurance Checklist

- [x] Commit history is devoid of any _merge commits_ and readable to facilitate reviews
- [x] If **new logic** ⚙️ is introduced: unit tests were added
- [x] If **new user stories** 🤏 are introduced: E2E tests were added
- [ ] If **new UI components** 🕹️ are introduced: corresponding stories in Storybook were created
- [ ] If **breaking changes** 🪚 are introduced: add the `breaking change` label
- [ ] If **bugs** 🐞 are fixed: add the `backport <release branch>` label
- [ ] The [documentation website](docs) 📚 has received the love it deserves

### CODE REVIEW

**Review by jahow — APPROVED** [review]
Thank you! The menu works and looks very good.

I think at some point we'll need to factorize all the menu opening code in a separate presentation component, but this can be done later on, especially when we'll want to do the same thing for the rollback confirmation dialog in the editor form:
image

**Comment by cmoinier** [comment]
> Thank you! The menu works and looks very good.
> 
> I think at some point we'll need to factorize all the menu opening code in a separate presentation component, but this can be done later on, especially when we'll want to do the same thing for the rollback confirmation dialog in the editor form: image

Right, also since the Import menu works in a very similar, it would make sense indeed. Let's keep that in mind!

File: libs/ui/search/src/lib/results-table/results-table.component.spec.ts [hunk 1]
```diff
@@ -158,19 +158,13 @@ describe('ResultsTableComponent', () => {
 
     beforeEach(() => {
       recordToBeDuplicated = null
-      component.duplicateRecord.subscribe((r) => (recordToBeDuplicated = r))
+      component.duplicateRecord.subscribe((r) => {
+        recordToBeDuplicated = r
+      })
     })
 
     it('emits a duplicateRecord event', () => {
```
[the remark this answers is not in the document]
Comment (jahow) [reply libs/ui/search/src/lib/results-table/results-table.component.spec.ts:166]: that makes sense :) although please keep in mind that sometimes unit tests can interact with the DOM too, if there's some logic related to DOM events for instance!

[the remark this answers is not in the document]
Comment (cmoinier) [reply libs/ui/search/src/lib/results-table/results-table.component.spec.ts:166]: Yes! In this case it seemed to me that it was something that could be done (and is already done, actually), in the e2e. I figured it made more sense to test the related logic without all the clicking on the fixture :)

**Pull request merged**